### PR TITLE
Main updates

### PR DESCRIPTION
**Pull request by JAWD-001**

### CODE REVIEW

File: peer_mentoring/groups/views.py [hunk 1]
```diff
@@ -52,6 +53,9 @@ def groups_moderated(request):
 def group_detail(request, group_id):
     group = get_object_or_404(Group, pk=group_id)
     posts = Post.objects.filter(group=group_id)
+    comment_count = Post.objects.filter(group=group_id).annotate(
```
Comment (bbelderbos) [inline peer_mentoring/groups/views.py:56]: this is correct, you can now use comment_count as attribute on the post, so that probably still needs fixing in the template as I don't see the count when adding 2 comments on a post ...

```
(Pdb) comment_count
<QuerySet [<Post: post1>]>
(Pdb) comment_count[0]
<Post: post1>
(Pdb) comment_count[0].comment_count
2
```

**Review by bbelderbos — CHANGES_REQUESTED** [review]

File: peer_mentoring/groups/views.py (hunk 1 above)
Comment (bbelderbos) [reply peer_mentoring/groups/views.py:56]: Please merge this in, did a quick PR with fix here: https://github.com/JAWD-001/peer_mentoring/pull/70

HTH
Bob

File: peer_mentoring/account_management/templates/profile_index.html [hunk 2]
```diff
@@ -13,7 +13,7 @@
       {% if profile.user != request.user %}
       {% if profile.user not in request.user.userprofile.friends.all %}
       <div>
-        <form method="post" id="friend_request" action="{% url 'account_management:profile_home' %}" style="display:inline;">
+        <form method="post" id="friend_request" action="{% url 'account_management:send_request' user.id %}" style="display:inline;">
```
Comment (bbelderbos) [inline peer_mentoring/account_management/templates/profile_index.html:16]: Was this the regression? Why this change?

Comment (JAWD-001) [reply peer_mentoring/account_management/templates/profile_index.html:16]: I made this change because the send request view should send the friend request. However, I initially had the profile home url cause i copied the url code within an href from a different file instead of rewriting and forgot to change.

File: peer_mentoring/account_management/templates/profile_index.html [hunk 3]
```diff
@@ -10,10 +10,10 @@
         <div class="fw-bold"><a href="{% url 'account_management:view_profile' profile.user.id %}">{{ profile.user.username }}</a></div>
         <p class="card-text">{{profile.user.title}}</p>
       </div>
-      {% if profile.user != request.user %}
-      {% if profile.user not in request.user.userprofile.friends.all %}
+      {% if profile != request.user %}
```
Comment (bbelderbos) [inline peer_mentoring/account_management/templates/profile_index.html:13]: are these objects the same, seems former is profile, latter is user

File: peer_mentoring/account_management/templates/profile_index.html [hunk 4]
```diff
@@ -10,10 +10,10 @@
...
+      {% if profile not in request.user.userprofile.friends.all %}
       <div>
-        <form method="post" id="friend_request" action="{% url 'account_management:send_request' user.id %}" style="display:inline;">
+        <form method="post" id="friend_request" action="{% url 'account_management:send_request' user_id=profile.user.id %}" style="display:inline;">
```
Comment (bbelderbos) [inline peer_mentoring/account_management/templates/profile_index.html:16]: nitpick (ignore for now) but you can use a bootstrap css class instead of `style="display:inline;"`

File: peer_mentoring/account_management/views.py [hunk 5]
```diff
@@ -64,15 +64,7 @@ def view_profile(request, user_id):
...
-    profiles = UserProfile.objects.all()
-    if request.method == "POST":
-        if "user_id" in request.POST:
-            user_id = request.POST.get("user_id")
-            user = get_object_or_404(UserProfile, id=user_id)
-            user.friends.add(user_id)
-            user.save()
-            messages.success(request, "Friend Request Sent")
-            return redirect("account_management:view_profile", user_id)
+    profiles = UserProfile.objects.all().exclude(user=request.user)
```
Comment (bbelderbos) [inline peer_mentoring/account_management/views.py:67]: I don't think you need all() before exclude()

Comment (bbelderbos) [reply peer_mentoring/account_management/views.py:67]: Good find and use 👍

File: peer_mentoring/account_management/views.py [hunk 6]
```diff
@@ -126,14 +118,16 @@ def accept_friend_request(request, request_id):
...
-    return redirect("account_manage:request_index")
+        messages.success(request, "Friend request rejected")
+    return redirect("account_management:request_index")
 
 
 def friend_request_index(request):
-    friend_requests = FriendRequest.objects.filter(receiver=request.user.userprofile)
+    friend_requests = FriendRequest.objects.filter(
+        receiver=request.user.userprofile
+    ).filter(accepted=False)
```
Comment (bbelderbos) [inline peer_mentoring/account_management/views.py:131]: can you do this with a single filter()?

File: peer_mentoring/chat/consumers.py [hunk 7]
```diff
@@ -28,6 +32,9 @@ async def receive(self, text_data):
                 "datetime": now.isoformat(),
             },
         )
+        message = ChatMessage.objects.create(
```
Comment (bbelderbos) [inline peer_mentoring/chat/consumers.py:35]: nice, happy you already started this!

File: peer_mentoring/chat/models.py [hunk 8]
```diff
@@ -0,0 +1,9 @@
+from django.contrib.auth.models import User
+from django.db import models
+
+
+class ChatMessage(models.Model):
+    id = models.BigAutoField(primary_key=True)
```
Comment (bbelderbos) [inline peer_mentoring/chat/models.py:6]: no need to specify the primary key, Django uses id by default

File: peer_mentoring/chat/models.py [hunk 9]
```diff
@@ -0,0 +1,9 @@
...
+    user = models.ForeignKey(User, on_delete=models.CASCADE)
+    message = models.TextField(max_length=300, blank=False, null=False)
+    added = models.DateTimeField(auto_now_add=True, null=False, blank=False)
```
Comment (bbelderbos) [inline peer_mentoring/chat/models.py:9]: I don't think you need `null=False, blank=False` here

**Review by bbelderbos — APPROVED** [review]
Great, few small things.
